src/lib/ketrew_configuration.ml

```diff
@@ -234,28 +234,21 @@ module File = struct
 
   let get_path ?root () =
     let env n () = try Some (Sys.getenv n) with | _ -> None in
+    let exists path () = if Sys.file_exists path then Some path else None in
     let try_options l last_resort =
       match List.find_map l ~f:(fun f -> f ()) with
       | Some s -> s
-      | None -> last_resort () in
-    let findout_path () =
+      | None -> last_resort in
+    let possible_files = [
+        env "KETREW_CONFIGURATION";
+        env "KETREW_CONFIG";
+      ] @ List.map (exists) default_configuration_filenames in
+    let ketrew_path =
       try_options [
         (fun () -> root);
         env "KETREW_ROOT";
-      ] (fun () -> default_ketrew_path) in
-    let find_in_path () =
-      let ketrew_path = findout_path () in
-      try_options
-        (List.map default_configuration_filenames
-           ~f:(fun path ->
-               fun () -> if Sys.file_exists path then Some path else None))
-        (fun () -> ketrew_path ^ "configuration.json")
-    in
-    try_options [
-      env "KETREW_CONFIGURATION";
-      env "KETREW_CONFIG";
-    ]
-      find_in_path
+      ] default_ketrew_path in
+    try_options possible_files (ketrew_path ^ "configuration.json")
```

Comment:
But the order in which tests are performed is very important for the priority of env-variables.
(I added the label `~and_then` to the fixed version):

https://github.com/hammerlab/ketrew/tree/fix-get-configuration-path